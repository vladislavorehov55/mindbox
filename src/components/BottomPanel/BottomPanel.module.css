.container {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
  padding: 5px 10px;
  background: #fff;
}
.btn {
  background-color: #fff;
  border: none;
  cursor: pointer;
  margin-left: 5px;
  padding: 5px;
}
.btnActive {
  border: 1px solid black;
}