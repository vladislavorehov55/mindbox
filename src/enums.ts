export enum TaskType {
  all = 'all',
  active = 'active',
  completed = 'completed'
}
export enum FilterType {
  all = 'all',
  active = 'active',
  completed = 'completed'
}