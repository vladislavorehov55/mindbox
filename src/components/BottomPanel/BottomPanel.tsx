import React, {FC, useCallback} from 'react';
import {FilterType} from '../../enums';
import styles from './BottomPanel.module.css';

type BottomPanelProps = {
  activeTasksCount: number,
  filteredType: FilterType,
  filterTasks: (filterType: FilterType) => void,
  clearCompleted: () => void
}
const BottomPanel: FC<BottomPanelProps> = ({activeTasksCount, filterTasks, clearCompleted, filteredType}) => {
  const btnArr = [
    {text: 'All', type: FilterType.all},
    {text: 'Active', type: FilterType.active},
    {text: 'Completed', type: FilterType.completed}
  ]
  const getBtnClasses = useCallback((type: FilterType) => {
    return type === filteredType ? `${styles.btn} ${styles.btnActive}` : styles.btn
  }, [filteredType])

  return (
    <div className={styles.container}>
      <span>{activeTasksCount} items left</span>
      <div>
        {
          btnArr.map((btn, ind) => {
            return <button key={ind} className={getBtnClasses(btn.type)} onClick={() => filterTasks(btn.type)}>{btn.text}</button>
          })
        }
      </div>
     <button className={styles.btn} onClick={clearCompleted}>Clear completed</button>
    </div>
  );
};

export default BottomPanel;