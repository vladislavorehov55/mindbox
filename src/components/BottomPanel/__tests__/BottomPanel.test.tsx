import React from 'react';
import {mount, ReactWrapper} from 'enzyme';
import BottomPanel from "../BottomPanel";
import {FilterType} from "../../../enums";

describe('BottomPanel component', () => {
    let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    const clearCompleted = jest.fn()
    const filterTasks = jest.fn()

    const getWrapper = () => mount(<BottomPanel clearCompleted={clearCompleted} filterTasks={filterTasks} activeTasksCount={1} filteredType={FilterType.all}/>);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = getWrapper();
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should call clearCompleted function', () => {
        const clearCompletedBtn = wrapper.find('button').at(3)
        clearCompletedBtn.simulate('click')
        expect(clearCompleted).toHaveBeenCalled()
    });
    it('should call filterTasks with correct options', () => {
        const allBtn = wrapper.find('button').at(0);
        allBtn.simulate('click');
        expect(filterTasks).toHaveBeenCalledWith(FilterType.all);

        const activeBtn = wrapper.find('button').at(1);
        activeBtn.simulate('click');
        expect(filterTasks).toHaveBeenCalledWith(FilterType.active);

        const completeBtn = wrapper.find('button').at(2);
        completeBtn.simulate('click');
        expect(filterTasks).toHaveBeenCalledWith(FilterType.completed)
    });
})