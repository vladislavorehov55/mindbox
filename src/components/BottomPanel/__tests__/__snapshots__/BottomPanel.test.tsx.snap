// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPanel component should render correctly 1`] = `
<BottomPanel
  activeTasksCount={1}
  clearCompleted={[MockFunction]}
  filterTasks={[MockFunction]}
  filteredType="all"
>
  <div>
    <span>
      1
       items left
    </span>
    <div>
      <button
        className="undefined undefined"
        key="0"
        onClick={[Function]}
      >
        All
      </button>
      <button
        key="1"
        onClick={[Function]}
      >
        Active
      </button>
      <button
        key="2"
        onClick={[Function]}
      >
        Completed
      </button>
    </div>
    <button
      onClick={[MockFunction]}
    >
      Clear completed
    </button>
  </div>
</BottomPanel>
`;
